### ISSUE TITLE
Tomcat war deployment version issue in new SALT version

### ISSUE THREAD

**Issue by pankajghadge** [saltstack/salt#30694]
Hi,

we have migrating our application to another environment & here our SALT version changed 

so we are facing war file version issue,  previously it was not there.

```
War file name: V6R2013xD5.war
Context Path: /ApplicationManger

```
In webapps folder it was deploying like this:

```
[root@app1 webapps]# ls
ApplicationManager##V6R2013xD6      manager
ApplicationManager##V6R2013xD6.war  ROOT     
[root@app1 webapps]#


```
but after SALT version change it is deploying like below:
```
[root@app1 webapps]# ls
ApplicationManager##NONE      manager
ApplicationManager##NONE.war  ROOT     
[root@app1 webapps]#


```  
 Is it possible to declare version in tomcat.war_deploy state?

After checking code I found that it was previously using base name but now it is checking for digit.
Can we tell salt to check for base name instead of digit outside of salt code?

`version = re.findall("-([\\d.-]+)$", os.path.basename(war).replace('.war', ''))
`
on more issue, here because of first "-" (hash) it is not checking for last digit. 

SALT version (salt 2014.7.1 (Helium)
 ```
# Prepare options
   opts = {
       'war': 'file:{0}'.format(tfile),
       'path': context,
       'version': os.path.basename(war).replace('.war', ''),
   }

```
/usr/lib/python2.7/site-packages/salt/modules/tomcat.py 
(salt-call 2015.8.3 (Beryllium)) 
```
def _extract_version(war):
    '''
    extract the version from the war name
    '''
    version = re.findall("-([\\d.-]+)$", os.path.basename(war).replace('.war', ''))
    return version[0] if len(version) == 1 else None

```

**Comment by jfindlay** [comment]
Related to #24520.

**Comment by jfindlay** [comment]
@pankajghadge, thanks for the report.  Unfortunately it doesn't seem possible to specify the war version independently of the war file.  This should be fixed in #30750.  Do you know if the war version is always the war filename minus the file suffix (`.war`)?
